### PR TITLE
Refactor Markdownz's handling of custom image dimensions

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
lib-react-components

## Linked Issue and/or Talk Post
Closes: https://github.com/zooniverse/front-end-monorepo/issues/2554

## Describe your changes

- Allow `height` to be undefined when parsing markdown image strings that contain custom dimensions such as "=200x" or "=100x100".
- Refactored Media component's stories to have more explicit examples of images
- linting

## How to Review

This can be reviewed by looking at the Media or Markdownz stories in lib-react-components storybook.

You can also run app-project locally and visit Superwasp's [Team page](https://local.zooniverse.org:3000/projects/hughdickinson/superwasp-black-hole-hunters/about/team?env=production). The network requests should not error, and the images in the page's markdown should be source from thumbnails.zooniverse.org.

Note that I will follow up with a documented Issue and separate PR for addressing how field guide icons are incorrectly passing dimensions to the Media component.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
- [ ] The PR creator has listed user actions to use when testing if bug is fixed
- [ ] The bug is fixed
- [ ] Unit tests are added or updated
## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected
(https://github.com/zooniverse/how-to-zooniverse/blob/master/Translations/lokalise.md#lokalise-and-fem).

### CODE REVIEW

File: packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js [hunk 1]
```diff
@@ -85,7 +84,5 @@ ThumbnailImage.propTypes = {
 }
 
 ThumbnailImage.defaultProps = {
```
Comment (eatyourgreens) [inline packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js:86]: This should be removed and replaced with default function args.

File: packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js [hunk 2]
```diff
@@ -31,19 +30,19 @@ function stringifySize (size) {
...
 
 export default function ThumbnailImage({
-  alt,
-  delay,
-  fit,
-  flex,
-  height,
-  origin,
-  placeholder,
+  alt, // ''
```
Comment (eatyourgreens) [inline packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js:33]:
```suggestion
  alt = defaultProps.alt, // ''
```

File: packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js [hunk 3]
```diff
@@ -31,19 +30,19 @@ function stringifySize (size) {
...
+  delay, // 0
```
Comment (eatyourgreens) [inline packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js:34]:
```suggestion
  delay = defaultProps.delay, // 0
```

File: packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js [hunk 4]
```diff
@@ -31,19 +30,19 @@ function stringifySize (size) {
...
+  fit, // cover
```
Comment (eatyourgreens) [inline packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js:35]:
```suggestion
  fit = defaultProps.fit, // cover
```

File: packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js [hunk 5]
```diff
@@ -31,19 +30,19 @@ function stringifySize (size) {
...
+  flex, // grow
```
Comment (eatyourgreens) [inline packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js:36]:
```suggestion
  flex = defaultProps.flex, // grow
```

File: packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js [hunk 6]
```diff
@@ -31,19 +30,19 @@ function stringifySize (size) {
...
+  height = DEFAULT_THUMBNAIL_DIMENSION,
+  origin, // https://thumbnails.zooniverse.org
```
Comment (eatyourgreens) [inline packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js:38]:
```suggestion
  origin = defaultProps.origin, // https://thumbnails.zooniverse.org
```

File: packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js [hunk 7]
```diff
@@ -31,19 +30,19 @@ function stringifySize (size) {
...
+  placeholder, //  null
```
Comment (eatyourgreens) [inline packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js:39]:
```suggestion
  placeholder = defaultProps.placeholder, //  null
```

**Review by eatyourgreens — CHANGES_REQUESTED** [review]
This looks good but `Component.defaultProps` is deprecated for functional components and should be replaced with default function args. 

The README and `package.json` should also be updated, and this should be published as a patch version update to `@zooniverse/react-components`.

**Comment by eatyourgreens** [comment]
If https://github.com/zooniverse/thumbnailer/pull/25 merges, you can drop the second parameter in the URL and resize only to width with a URL like `/400x/`. That will allow us to support resizing of transcription subjects. At the moment, portrait images aren't properly resized when the max height is 999. That causes some layout problems for transcription projects in PFE.

File: packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js (hunk 1 above)
Comment (goplayoutside3) [reply packages/lib-react-components/src/Media/components/ThumbnailImage/ThumbnailImage.js:86]: Good point! I'll add default function args instead.

**Comment by goplayoutside3** [comment]
> The README and package.json should also be updated with a new patch version, and this should be published as a patch to @zooniverse/react-components.

Good reminder - Once this PR is merged I'll follow-up with an update to the Changelog and publish the new version

**Pull request merged**